### ISSUE TITLE
Pd hangs if it can't create docspath.conf (read-only filesystem)

### ISSUE THREAD

**Issue by jeremiahrose** [pure-data/pure-data#669]
Hi there,
I am using Pd-0.49.0 and trying to get it to work in my custom Buildroot OS that features a read-only root filesystem.
I have it compiled and playing a test tone with the -nogui flag. However, with the -nogui flag removed, the app simply hangs without producing any sound or showing the GUI.
Interestingly, if I rebuild my OS with a writeable root filesystem, the Puredata TK GUI works absolutely fine.
So, it seems that Pd has issues running on a read-only FS, and I would like to try and fix this. I'm wondering if anyone has a list of files that Pd or it's Wish GUI needs to open or write to in order to work correctly?

**Comment by umlaeute** [comment]
i used to be able to run Pd on read-only filesystems.

two things come to my mind that might have introduced problems:
- the recent addition of Pd-Documents
- deken, the package manager

please try to get a list of all files that are being opened during the startup, and see where it hangs. something like:

~~~sh
strace -o pd-gui.strace  -e trace=%file wish /path/to/tcl/pd-gui.tcl
~~~

(it's important to start Pd via pd-gui (rather than via "pd", which is the core), if we want to see where the GUI-process hangs)

**Comment by jeremiahrose** [comment]
By using `strace` on a r/w filesystem and removing write permissions on directories one by one, it seems like my home directory is the issue. In particular, if I delete the `~/.config/Pd/org.puredata.pd.pd-gui/docspath.conf` file and then make `~/` unwriteable, `pd` will not produce a GUI.
Is this the recently introduced Pd-Documents you're referring to? Do you have any suggestions (other than symlink the file to a tmpfs, which is what I will do for now) on how to configure pd to create this file elsewhere, or to do without?

**Comment by jeremiahrose** [comment]
Expected behaviour is that is would print an error message `Warning: Can not create docspath.conf: read-only filesystem` and then run anyway without it.

**Comment by umlaeute** [comment]
turned out it was a problem with the error-reporting of the GUI-preference system (docsdir triggered it, but was totally innocent).

so everything was implemented as suggested, only the reporting of the problem would crash :-)
it's fixed now in the `update/0.50` branch (which is supposed to be merged into master before the next release)

btw, a slightly simpler way to reproduce the problem is:
~~~sh
mkdir /tmp/pd-ro; chmod a-w /tmp/pd-ro; HOME=/tmp/pd-ro ./bin/pd
~~~

and as a workaround for the problem, you can also tell Pd to use a writable filesystem for It's preferences, as it respects both the `XDG_CONFIG_HOME` and the `PD_CONFIG_DIR` env-vars.

**Comment by jeremiahrose** [comment]
Thanks Umlaeute for the fix and workaround!
